install_config/storage_examples/containerized_heketi_with_dedicated_gluster.adoc

```diff
@@ -0,0 +1,386 @@
+[[install-config-storage-examples-containerized-heketi-dedicated-gluster]]
+= Example: Containerized Heketi for managing dedicated GlusterFS storage
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+[[containerized-heketi-dedicated-gluster-overview]]
+== Overview
+This example provides information about the integration, deployment, and management of GlusterFS containerized storage nodes by using Heketi running on OpenShift Container Platform.
+
+This example:
+
+* Shows how to install and configure a Heketi server on OpenShift to perform dynamic provisioning.
+* Assumes you have familiarity with Kubernetes and the Kubernetes Persistent Storage model.
+* Assumes you have access to an existing, dedicated GlusterFS cluster that has raw devices available for consumption and management by a Heketi server. If you do not have this, you can create a three node cluster using your virtual machine solution of choice. Ensure sure you create a few raw devices and give plenty of space (at least 100GB recommended). See
+link:https://access.redhat.com/documentation/en-US/Red_Hat_Storage/3.1/html/Installation_Guide/[Red Hat Gluster Storage Installation Guide].
+
+[[containerized-heketi-dedicated-gluster-prerequisites]]
+== Environment and Prerequisites
+This example uses the following environment and prerequisites:
+
+* GlusterFS cluster running Red Hat Gluster Storage (RHGS) 3.1. Three nodes, each with at least two 100GB RAW devices:
+** *gluster23.rhs* (192.168.1.200)
+** *gluster24.rhs* (192.168.1.201)
+** *gluster25.rhs* (192.168.1.202)
+* This example uses an all-in-one {product-title} cluster (master and node on a single host), though it can work using a standard, multi-node cluster as well.
+** *k8dev2.rhs* (192.168.1.208)
+
+[[containerized-heketi-dedicated-gluster-install-heketi]]
+== Installing and Configuring Heketi
+Heketi is used to manage the Gluster cluster storage (adding volumes, removing volumes, etc.). Download link:https://github.com/heketi/heketi/blob/master/extras/openshift/templates/deploy-heketi-template.json[`deploy-heketi-template`] to install Heketi on OpenShift.
+
+[NOTE]
+====
+This template file places the database in an EmptyDir volume. Adjust the database accordingly for a reliable persistent storage.
+====
```

Comment:
Does `EmptyDir` need mark up? i looked quickly at the docs and there doesn't seem to be consistent usage. Some `EmptyDir`, some *EmptyDir* and some with no mark up.

Reply:
Kubernetes uses `emptyDir` maybe we should use the same markup. I'll check this in our docs and submit a new PR with the changes.

Reply:
Thanks @mburke5678 addressed this in https://github.com/openshift/openshift-docs/pull/6084